drivers/Watchdog.cpp

```diff
@@ -20,6 +20,72 @@
 
 namespace mbed {
 
+uint32_t Watchdog::_is_initialized = 0;
+uint32_t Watchdog::_re_initialize = 0;
+uint32_t Watchdog::_bitmask = 0;
+uint32_t Watchdog::_kick_bitmask = 0;
+Watchdog::watchdog_usr_info Watchdog::_usr_info[MAX_THREAD_WATCHDOG_SUPPORT] = {0};
+rtos::Mutex Watchdog::_ThreadSafeLockMutex;
+mbed_error_status_t Watchdog::wd_register(const osThreadId_t tid, const uint32_t timeout)
```

Comment:
* No need for `wd_` prefix here
* Can you remove the `tid` parameter and use the current thread id instead? This will also provide us with a migration path for bare-metal support
* No need for `timeout` to be `const`